### PR TITLE
Add CloudWatch Logs Integration

### PR DESCRIPTION
**Pull request by confusdcodr**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,7 +1,48 @@
...
+  create_log_group_role     = var.cloud_watch_logs_role_arn == null
+  cloud_watch_logs_role_arn = local.create_log_group_role ? aws_iam_role.this[0].arn : var.cloud_watch_logs_role_arn
+}
+
+### RESOURCES ###
+# Create CloudWatch Log Group
+resource "aws_cloudwatch_log_group" "this" {
+  count = local.create_log_group ? 1 : 0
+
+  name              = "Tardigrade/Cloudtrail/${var.cloudtrail_name}"
```
Comment (userhas404d) [inline main.tf]:
```suggestion
  name              = local.cloud_watch_logs_group_name
```

File: main.tf [hunk 2]
```diff
@@ -31,3 +75,46 @@ resource "aws_cloudtrail" "this" {
...
+
+### DATA SOURCES ###
+data "aws_partition" "current" {}
+
+data "aws_region" "current" {}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_cloudwatch_log_group" "this" {
+  count = var.cloud_watch_logs_group_name == null ? 0 : 1
```
Comment (userhas404d) [inline main.tf]:
```suggestion
  count = local.create_log_group ? 0 : 1
```

File: .bumpversion.cfg [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 2.0.0
+current_version = 2.0.1
```
Comment (lorengordon) [inline .bumpversion.cfg]: This is a new feature, so `bumpversion minor`, not patch...

File: main.tf [hunk 4]
```diff
@@ -1,7 +1,48 @@
-provider "aws" {
+provider "aws" {}
+
+### LOCALS ###
+locals {
+  create_log_group            = var.cloud_watch_logs_group_name == null
+  cloud_watch_logs_group_name = local.create_log_group ? "Tardigrade/Cloudtrail/${var.cloudtrail_name}" : var.cloud_watch_logs_group_name
```
Comment (lorengordon) [inline main.tf]: Is there a log group name that AWS uses for this, by default, when going through the console or anything? i.e. something like `/aws/cloudtrail/<name>`? we usually try to line up with what the user would see if they used the AWS interfaces themselves...

Comment (confusdcodr) [reply main.tf]: I tried to create a log group with the `aws` prefix and it is denied stating that it is reserved for AWS. If I try to create one via CloudTrail, it defaults to `CloudTrail/DefaultLogGroup`

File: main.tf (hunk 1 above)
Comment (confusdcodr) [reply main.tf]: Fixed

File: main.tf (hunk 2 above)
Comment (confusdcodr) [reply main.tf]: Fixed

**Comment by confusdcodr** [comment]
@lorengordon - The CloudTrail resource takes a log group and role in order to write the events into CloudWatch

File: main.tf (hunk 4 above)
Comment (lorengordon) [reply main.tf]: `/aws/...`, the leading `/` seems to be important. anyway, interesting. apparently aws is not consistent about their prefixing like that. perhaps just newer services are using `/aws/<service>/` and all lowercase.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,17 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).
 
+### 2.0.1
```
Comment (lorengordon) [inline CHANGELOG.md]: the version needs to be updated here also

File: main.tf (hunk 4 above)
Comment (confusdcodr) [reply main.tf]: Playing with this in the cosole a little more... `/aws/<service>` seems to be fine. If you start with `aws` without the leading `/` then there is an error

Comment (confusdcodr) [reply main.tf]: Fixed to be `/aws/cloudtrail/<name>`

**Pull request merged**